geo/src/algorithm/geodesic_intermediate.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{CoordFloat, Point};
+use geographiclib_rs::{DirectGeodesic, Geodesic, InverseGeodesic};
+
+/// Returns a new Point along a route between two existing points on an ellipsoidal model of the earth
+
+pub trait GeodesicIntermediate<T: CoordFloat> {
+    /// Returns a new Point along a route between two existing points on an ellipsoidal model of the earth
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[macro_use] extern crate approx;
+    /// #
+    /// use geo::algorithm::geodesic_intermediate::GeodesicIntermediate;
+    /// use geo::Point;
+    ///
+    /// let p1 = Point::<f64>::new(10.0, 20.0);
+    /// let p2 = Point::<f64>::new(125.0, 25.0);
+    /// let i20 = p1.geodesic_intermediate(&p2, 0.2);
+    /// let i50 = p1.geodesic_intermediate(&p2, 0.5);
+    /// let i80 = p1.geodesic_intermediate(&p2, 0.8);
+    /// let i20_should = Point::new(29.842907, 29.951445);
+    /// let i50_should = Point::new(65.879360, 37.722253);
+    /// let i80_should = Point::new(103.556796, 33.506196);
+    /// assert_relative_eq!(i20.x(), i20_should.x(), epsilon = 1.0e-6);
+    /// assert_relative_eq!(i20.y(), i20_should.y(), epsilon = 1.0e-6);
```

Comment:
Thanks to @martinfrances107 in #567, points implement some `approx` traits, so you should be able to write this more succinctly:

```
assert_relative_eq!(i20, i20_should, epsilon = 1.0e-6)
```

Here and in the tests below.

Reply:
Oh, that's very nice